includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -978,22 +971,23 @@ public function process_payment_for_order( $cart, $payment_information, $additio
 			}
 
 			if ( empty( $intent ) ) {
-				// Create intention, try to confirm it & capture the charge (if 3DS is not required).
-				$intent = $this->payments_api_client->create_and_confirm_intention(
-					$converted_amount,
-					$currency,
-					$payment_information->get_payment_method(),
-					$customer_id,
-					$payment_information->is_using_manual_capture(),
-					$save_payment_method_to_store,
-					$payment_information->should_save_payment_method_to_platform(),
-					$metadata,
-					$this->get_level3_data_from_order( $order ),
-					$payment_information->is_merchant_initiated(),
-					$additional_api_parameters,
-					$payment_methods,
-					$payment_information->get_cvc_confirmation()
-				);
+				add_filter( 'woopay_create_and_confirm_intention_request_from_base_intent_request', [ Checkout_Service::class, 'create_woopay_intention_request' ], 10, 3 );
```

Comment:
I'm fine leaving this line here within this PR, but in the long term it should be moved within `WooPay\Checkout_Service` as well.

Reply:
Also, where does `woopay_create_and_confirm_intention_request_from_base_intent_request` from? Did you test this in any way? It's not being applied within manual tests (such a filter is not called anywhere), and I don't see it in tests.

Reply:
This is addressed in this commit: https://github.com/Automattic/woocommerce-payments/pull/5136/commits/6df27962755b9e06dcc5325a13d9ded7a211fc09 . The functions are moved to WooPay Checkout service. If we agree on this approach, I will replace/fix  all duplicate occurrences